lib/pulselistener.js

```diff
@@ -112,7 +140,21 @@ PulseConnection.prototype.connect = function() {
   // Connect if we're not already doing this
   if (!this._connecting) {
     this._connecting = true;
-    retryConnect(this._connectionString, 7).then(function(conn) {
+    if (this.isTaskclusterPulse) {
+      // Request new credentials every time we create a connection in case they
+      // have been rotated.
+      var getCredentials = this._pulse.namespace(
+        this.namespace, {contact: this._contact}).then(function(credentials) {
+          that._connectionString = buildPulseConnectionString({
+            username: credentials.username,
+            password: credentials.password,
+            hostname: that.hostname,
+          });
+        });
+    }
+    Promise.resolve(getCredentials).then(function() {
```

Comment:
You should declare `getCredentials` sooner,  so it's not out of scope when you use it...

In some nice future we want to use `let` instead of `var`